### PR TITLE
Refactor transak stuff and make code more funcitonal

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
The feature/transak_integration was great, but it was implemented by Object oriented approach and it is too class based,  so I created this PR to make it functional and simplify the code

### CODE REVIEW

File: resolvers/donationResolver.ts [hunk 1]
```diff
@@ -135,8 +135,8 @@ export class DonationResolver {
         createdAt: new Date(),
         toWalletAddress: toAddress.toString().toLowerCase(),
         fromWalletAddress: fromAddress.toString().toLowerCase(),
-        transakId: transakId,
-        transakStatus: transakStatus,
```
Comment (mohammadranjbarz) [inline resolvers/donationResolver.ts:139]: Is this safe to get transakStatus from user?
user can send `COMPLETED` while creating
I suggest we should  either set the transak status ourself or don't set in saveDonation mutation

Comment (CarlosQ96) [reply resolvers/donationResolver.ts:139]: This is a posible outcome. We should probably skip the initial status and set it ourselves.

Comment (mohammadranjbarz) [reply resolvers/donationResolver.ts:139]: Ok, so I will change it not save in creating a donation
but I don't change the query Arg to our query be backward compatible for frontend

Comment (CarlosQ96) [reply resolvers/donationResolver.ts:139]: I also think we shouldn't notify segment initially if it's a transak Donation as it's never in the completed status the first time.

File: services/transak/donationService.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    donation.amount = transakData.webhookData.cryptoAmount
+  }
+
+  if (donation.toWalletAddress.toLowerCase() !== transakData.webhookData.walletAddress.toLowerCase()){
+    donation.toWalletAddress= transakData.webhookData.walletAddress
+    const project = await Project.findOne({
+      walletAddress: transakData.webhookData.walletAddress
+    })
+    // we should check the walletAddress is matched with what is in donation, ir prevents fraud
+    donation.projectId = project?.id || 0
```
Comment (CarlosQ96) [inline services/transak/donationService.ts]: What happens if they don't match? should we raise an exception?

Comment (mohammadranjbarz) [reply services/transak/donationService.ts]: Currently I think we can 0 for projectId, in the future we can decide about that

**Pull request merged**